### PR TITLE
UI for YPP authorization flow

### LINKED ISSUE #3153: UI for YPP authorization flow
**Scope:**

- [ ] Implement all the UI and forms used in YPP authorization flow - https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=1774%3A115618

**OOP:**

- [ ] Google authentication
- [ ] YPP backend
- [ ] YouTube criteria validation

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #3153 

You can test it using storybook under `/?path=/story/ypp-yppauthorizationstepmodal--default`

**Design**
https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=1774%3A115631

### CODE REVIEW

File: packages/atlas/src/components/_ypp/DetailsFormDialog/DetailsFormDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                }
+                return true
+              },
+            },
+            onChange: (event) => {
+              const { value } = event.target
+              setLoading(true)
+              setValue('channel', event.target.value, { shouldTouch: true, shouldDirty: true })
+              if (value.length) {
+                debounceFetchChannels.current(value)
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/DetailsFormDialog/DetailsFormDialog.tsx]: How this validation actually works? I think errors just randomly show up, and doesn't matter if the channel exists or not

https://user-images.githubusercontent.com/51168865/190166749-81e6b132-cd06-45a1-ba31-830a95c6a417.mov

<shameless_plug>I think you could look at the code in the create membership form. The debounced validation works pretty well there </shameless_plug>

File: packages/atlas/src/components/_ypp/DetailsFormDialog/DetailsFormDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }, 500)
+  )
+  return (
+    <FormFieldsWrapper>
+      <FormField
+        label="Contact email"
+        description="We need your email address to send you payment information. No spam or marketing materials."
+        error={errors.email?.message}
+      >
+        <Controller
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/DetailsFormDialog/DetailsFormDialog.tsx]: Why do we need a controller here? We could use `register` as well if I'm not mistaken.

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      show={show}
+      dividers
+      primaryButton={authorizationStep?.primaryButton}
+      secondaryButton={{ text: !isSummary ? 'Back' : 'Close', onClick: onBackClick }}
+      additionalActionsNode={
+        isConnectWithYt ? (
+          <Button variant="tertiary">Learn more</Button>
+        ) : (
+          !isSummary && (
+            <Button variant="tertiary" onClick={onCancelClick}>
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: On the select channel, the Cancel button shouldn't be present.

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+          title: 'Terms & conditions',
+          description:
+            'Once automatic YouTube videos sync is available, in order for it to work, your Atlas channel [NEEDS TO DO WHAT?]. This is purely a technical measure and does not affect ownership and rights to the content uploaded to you Atlas channel.',
+          primaryButton: { text: 'Accept terms & sign' },
+          additionalSubtitleNode: (
+            <AdditionalSubtitle>
+              <Text variant="h400" as="span">
+                Automatic YouTube sync
+              </Text>{' '}
+              <Text variant="t100" as="span" color="colorText">
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: This should be `textMuted`
image

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        }
+      case 'terms-and-conditions':
+        return {
+          title: 'Terms & conditions',
+          description:
+            'Once automatic YouTube videos sync is available, in order for it to work, your Atlas channel [NEEDS TO DO WHAT?]. This is purely a technical measure and does not affect ownership and rights to the content uploaded to you Atlas channel.',
+          primaryButton: { text: 'Accept terms & sign' },
+          additionalSubtitleNode: (
+            <AdditionalSubtitle>
+              <Text variant="h400" as="span">
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: I could be wrong but I think the gap between Automatic Youtube Sync and dialog title is bigger than 24px. 
Check the margin of the dialog title: 
image

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                .
+              </Text>
+            </>
+          ),
+          primaryButton: { text: 'Go to dashboard' },
+          component: <Img src={appScreenshot} />,
+        }
+      case 'connect-with-yt':
+        return {
+          title: 'Connect with YouTube?',
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: There is some unnecessary space in this dialog: 
image

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+          description:
+            'Reupload and backup your YouTube videos to receive to receive a guaranteed payout in the YouTube Partner Program.',
+          primaryButton: {
+            text: 'Authorize with YouTube',
+          },
+        }
+    }
+  }, [channels, step])
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: Text in the secondary button should be `Close` not `Back`. The icons in the dialog are different as well: 
image

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx (hunk 3 above)
Comment (rafalpawlow) [reply packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: U sure? https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=1637%3A121102

Comment (drillprop) [reply packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: Sorry, I meant the `Back` button
image

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx (hunk 5 above)
Comment (rafalpawlow) [reply packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: I checked it in photoshop and it seems it's exactly `24px`
image

File: packages/atlas/src/components/_ypp/SelectChannelDialog/SelectChannelDialog.styles.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import styled from '@emotion/styled'
+
+export const ListItemsWrapper = styled.div`
+  margin: 0 calc(-1 * var(--local-size-dialog-padding));
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/SelectChannelDialog/SelectChannelDialog.styles.ts]: Unfortunately, the padding is not correct here. It should be 24px
Yours: 
image
Design:
image
image

**Review by kdembler — COMMENTED** [review]
You named all the steps `*Dialog` (for example `RequirementsDialog`) but they aren't actually dialogs, they are just steps content. Let's use the same scheme we have for `SignInModal` so:
- All the steps are called `YppAuthorization*Step`
- All the steps are placed in `YppAuthorizationSteps` directory within `YppAuthorizationStepModal` (I would actually suggest to drop the "Step" from this name for consistency)

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+          primaryButton: {
+            text: 'Continue',
+          },
+          component: <DetailsFormDialog />,
+        }
+      case 'terms-and-conditions':
+        return {
+          title: 'Terms & conditions',
+          description:
+            'Once automatic YouTube videos sync is available, in order for it to work, your Atlas channel [NEEDS TO DO WHAT?]. This is purely a technical measure and does not affect ownership and rights to the content uploaded to you Atlas channel.',
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: NEEDS TO DO WHAT? 
image

Let's add a todo to fix this copy, or replace it with some proper text.

File: packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+          title: 'Congratulations!',
+          description: (
+            <>
+              Congratulations! You just received 200 JOY as your Atlas channel is now officially enrolled in the YouTube
+              Partner Program and tied with a YouTube channel.{' '}
+              <Text variant="t200" as="p" margin={{ top: 2 }} color="colorText">
+                All information around your activity in the program can be found in the{' '}
+                <Button variant="primary" _textOnly>
+                  YPP page in Studio
+                </Button>
```
Comment (drillprop) [inline packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx]: We can add a link to `studio/ypp` here already

File: packages/atlas/scripts/figma-import/figma-import.config.js.example [hunk 11]
```diff
@@ -9,6 +9,6 @@
 module.exports = {
   FIGMA_TOKEN: '',
   FILE_KEY: '1uxugLps6oooI38PGmdrn9',
-  FRAME_WITH_ICONS_ID: '637:380',
-  FRAME_WITH_ILLUSTRATIONS_ID: '1008:3A6558'
+  FRAME_WITH_ICONS_ID: '1547:4330',
+    FRAME_WITH_ILLUSTRATIONS_ID: '1008:6558'
```
Comment (drillprop) [inline packages/atlas/scripts/figma-import/figma-import.config.js.example]:
```suggestion
  FRAME_WITH_ILLUSTRATIONS_ID: '1008:6558'
```
redundant indentation

**Review by drillprop — CHANGES_REQUESTED** [review]

**Review by drillprop — APPROVED** [review]
LGTM

**Review by kdembler — APPROVED** [review]

**Review by kdembler — APPROVED** [review]

**Pull request merged**